cardano-sync/src/Cardano/Sync/Plugin.hs

```diff
@@ -36,8 +38,8 @@ data SyncNodePlugin = SyncNodePlugin
     -- This will not be called for the original genesis block, but will be called for
     -- all subsequent blocks.
     -- Blocks (including epoch boundary blocks) are called in sequence from the oldest to the newest.
-  , plugInsertBlock :: [Trace IO Text -> SyncEnv -> [BlockDetails] -> IO (Either SyncNodeError ())]
-
+  , plugInsertBlock :: Trace IO Text -> SyncEnv -> [CardanoBlock] -> ExceptT SyncNodeError IO [BlockDetails]
```

Comment:
Why? This seems to reduce the generality of the "plugin" system.

Reply:
The idea is that `plugInsertBlock` also creates the `[BlockDetails]`. `plugInsertBlockDetails` makes sure that we don't reduce the generality of the "plugin" system.

Reply:
What is the difference between `plugInsertBlock` and `plugInsertBlockDetails` ?

Seems the only difference is `CardanoBlock` vs `BlockDetails`. Can these be used interchangably? Can we do without the older version?

Reply:
`plugInsertBlock` applies the block to the ledger state and creates the `BlockDetails`. So `plugInsertBlockDetails` has the details available to do any extra work with them.

Reply:
Ah ok, now I get it. I will add some comments to make that more clear.

Reply:
Maybe `plugInsertBlock` should even be renamed. Maybe something like `pluginPrepareBlocks` .